Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Builder
+FROM node:10-alpine as builder
+
+# Support custom branches of the react-sdk and js-sdk. This also helps us build
+# images of riot-web develop.
+ARG USE_CUSTOM_SDKS=false
+ARG REACT_SDK_REPO="https://github.com/matrix-org/matrix-react-sdk.git"
+ARG REACT_SDK_BRANCH="master"
+ARG JS_SDK_REPO="https://github.com/matrix-org/matrix-js-sdk.git"
+ARG JS_SDK_BRANCH="master"
+
+RUN apk add --no-cache git dos2unix
+
+WORKDIR /src
+
+COPY . /src
+RUN dos2unix /src/scripts/docker-link-repos.sh && sh /src/scripts/docker-link-repos.sh
```

Comment:
It makes no sense to convert it during the docker build. This should be done before adding it to the repository.

Reply:
It is committed with the right line endings, but Windows tends to not care once it is on disk.

Reply:
In this case it's not Windows but git on Windows that has automatic line conversion enabled. This should be fixable by telling git to not modify them:

```
# .gitattributes
*.sh		text eol=lf
```

For details see: https://git-scm.com/docs/gitattributes

Reply:
I've tried everything in the book, trust me.